src/NAR/RuntimeSystem.java

```diff
@@ -0,0 +1,73 @@
+package NAR;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class RuntimeSystem {
+	
+	private static RuntimeSystem Instance = new RuntimeSystem();
+	RuntimeSystem() {}
+	
+	
+	private List listOfLines(String str) {
+
+		List<String> lines = new ArrayList<>();
+		Scanner scanner = new Scanner(str);
+
+		while (scanner.hasNextLine()) {
+
+			lines.add(scanner.nextLine());
+		}
+
+		scanner.close();
+		return lines;
+	}
+
+	private String reduceSpace(String str) {
+
+		String line = str.replaceAll("\\s +", " ");
+
+		if (isLengthGreaterThanZero(line)) {
+
+			if (isFirstCharWhiteSpace(line)) {
+
+				line = line.replaceFirst("\\ +", "");
+			}
+		}
+
+		return line;
+	}
+
+	private static boolean isFirstCharWhiteSpace(String str) {
+
+		char fChar = str.charAt(0);
+		return Character.isWhitespace(fChar);
+	}
+
+	private static boolean isLengthGreaterThanZero(String str) {
+
+		return str != null && str.length() > 0;
+	}
+
+
+	public void runCode(String str) {
+
+		List<String> lines = Instance.listOfLines(str);
+
+		for (String line : lines) {
+
+			String statement = Instance.reduceSpace(line);
```

Comment:
given that reduceSpace and isLengthGreaterThanZero are static functions you should not use the Instance variable to access them. note that java will let you do that but it's improper